### DISCUSSION TITLE
Problems getting touch to work (Arduino RP2040, SPI TFT and Touch)

### DISCUSSION THREAD

**Discussion by ruifig** [Bodmer/TFT_eSPI#1530] [General]
Hi,

I'm having some problems getting TFT_eSPI 's touch code work, or any other touch library for that matter, if I'm actually drawing something with TFT_eSPI.

From the little I understand so far, it might be something related to SPI.
For example, if I run the "calibrateTouch" function in my setup, it never waits for me to touch anything, because `validateTouch` will succeed with some crazy `z` values, in the range 62000-64000.
```
    for(uint8_t j= 0; j<8; j++){
      // Use a lower detect threshold as corners tend to be less sensitive
      while(!validTouch(&x_tmp, &y_tmp, Z_THRESHOLD/2));  <<<<<<<<<<<  Always succeeds
      values[i*2  ] += x_tmp;
      values[i*2+1] += y_tmp;
      }
 ```

I messed around trying to get another touch library working (https://github.com/PaulStoffregen/XPT2046_Touchscreen), but was hitting similar problems.
Then I tried NOT drawing anything in my loop, and suddenly my touch values I was writing in my loop were correct.
In other words, if I mix drawing and touch, it doesn't work.

To troubleshoot it, I used the same wiring, but running Adafruit_ILI9341 for graphics and that touch library, and everything works fine.
I suspect might be something to do with the way TFT_eSPI is handling SPI. Maybe again related to Arduino Mbed vs Earle Philhower's packages, like in my other solved issue (https://github.com/Bodmer/TFT_eSPI/discussions/1519).

At the moment I'm clueless. :)

**Comment by Bodmer** [comment]
I assume you are still using the Arduino RP2040 Connect and ILI9241 screen.

Which RP2040 board package are you using (Earle Philhower's or the Arduino Mbed one)?

Have you tried a Raspbery Pi Pico?

**Comment by Bodmer** [comment]
FYI I created an alert here to help other users:
https://github.com/Bodmer/TFT_eSPI/discussions/1533
